# WeChat Pay Data Analysis
    * This Project will show my life process. My bill can represent the situation I get in to. Something like fall in love, Breaking up ETC.
    

In [104]:
import pandas as pd
import numpy as np

df = pd.read_csv("微信支付账单.csv")
print(df)

                     交易时间      交易类型    交易对方                 商品 收/支     金额(元)  \
0     2021-09-29 10:31:14        转账  Edward          转账备注:微信转账  支出  ¥1000.00   
1     2021-09-28 13:08:47      微信红包    小庄👼🏻                  /  收入   ¥188.00   
2     2021-09-28 13:08:42      微信红包    小庄👼🏻                  /  收入   ¥188.00   
3     2021-09-28 13:08:39      微信红包    小庄👼🏻                  /  收入   ¥188.00   
4     2021-09-28 10:48:16     转账-退款       /          转账备注:微信转账  收入   ¥100.00   
...                   ...       ...     ...                ...  ..       ...   
1227  2021-01-02 02:44:18  微信红包（单发）   发给·Q·                  /  支出    ¥66.66   
1228  2021-01-01 21:29:34      商户消费  杭州西溪天街  杭州西溪天街 浙A0392W停车费  支出    ¥30.00   
1229  2021-01-01 21:20:58      商户消费    搜电科技              充电宝租借  支出     ¥4.50   
1230  2021-01-01 18:22:57        转账     何骡子          转账备注:微信转账  支出    ¥60.00   
1231  2021-01-01 18:04:28        转账      粉红          转账备注:微信转账  支出    ¥65.00   

     支付方式   当前状态                       

### I don't need something in the dataframe becuase they are meaningless to present the result from my bill
    * We do not need to know order number. That is not my issue but the IT company's. They have to store that data to check their income and outcome.
    * Situation of the order. I never have faild payment so the situation of orders always paying successfully.
    * Comment is not nececary because through WeChat Pay, I do not use the comment but in alipay I would. But alipay's bill file has lots of problem.
#### That is why I drop these 4 columns.

In [105]:
df = df.drop(columns=['交易单号','商户单号','当前状态','备注'])
df["金额(元)"] = df["金额(元)"].map(lambda x: str(x).lstrip('¥').rstrip('¥')).astype(float)
df["交易时间"] = pd.to_datetime(df["交易时间"],format='%Y-%m-%d') 
#print(df["金额(元)"].sum())
df_income = df.loc[df['收/支'] == '收入']
df_outcome = df.loc[df['收/支'] == '支出']
print(df_income)

                    交易时间   交易类型  交易对方         商品 收/支   金额(元) 支付方式
1    2021-09-28 13:08:47   微信红包  小庄👼🏻          /  收入  188.00    /
2    2021-09-28 13:08:42   微信红包  小庄👼🏻          /  收入  188.00    /
3    2021-09-28 13:08:39   微信红包  小庄👼🏻          /  收入  188.00    /
4    2021-09-28 10:48:16  转账-退款     /  转账备注:微信转账  收入  100.00   零钱
6    2021-09-28 01:52:00     转账    妈👩  转账备注:生日快乐  收入  520.00    /
...                  ...    ...   ...        ...  ..     ...  ...
1193 2021-01-09 11:06:16     转账    妈👩  转账备注:微信转账  收入   50.00    /
1204 2021-01-07 14:23:50     转账    老姐  转账备注:微信转账  收入   50.00    /
1205 2021-01-07 11:32:17   微信红包    十一          /  收入    0.01    /
1214 2021-01-06 11:30:11   微信红包    十一          /  收入    0.03    /
1218 2021-01-05 19:22:26     转账   FFF  转账备注:微信转账  收入  500.00    /

[243 rows x 7 columns]


## Last year happened lots of things. 
    * I broke up 2 times last year 
    * First time I study abroad So I get more money from my family than before.

In [106]:
print(df_income['金额(元)'].sum())
df_ex1 = df_income.loc[df_income['交易对方'] == '清和月五']
df_ex2 = df_income.loc[df_income['交易对方'] == 'T__a']
df_ex1o = df_outcome.loc[df_outcome['交易对方'] == '清和月五']
df_ex2o = df_outcome.loc[df_outcome['交易对方'] == '.']
df_mom = df_income.loc[df_income['交易对方'] == '妈👩']
df_dad = df_income.loc[df_income['交易对方'] == '爸👨']
df_momo = df_outcome.loc[df_outcome['交易对方'] == '妈👩']
df_dado = df_outcome.loc[df_outcome['交易对方'] == '爸👨']


79406.96


## So I totally get 79406.96 rmb last year.
    * It can be converted to $12310

In [107]:
print(df_outcome['金额(元)'].sum())

84432.79000000001


# But I spend 84433 rmb last year
    * It can be converted to $13090
    * That is meaning I did not save any money last year

In [108]:
df_xiaofei = df_outcome.loc[df_outcome['交易类型'] == '商户消费']
print(df_xiaofei['金额(元)'].sum())
print((66576.44/84433)*100,"%")

66576.44


## I almost spend money on reatraunt and parking lot
    * I spend 66576 rmb on these things 

In [109]:
print("income from 1st ex",df_ex1['金额(元)'].sum())
print("income from 2st ex",df_ex2['金额(元)'].sum())
print("income from dad and mom",df_mom['金额(元)'].sum() + df_dad['金额(元)'].sum())
print("outcome to 1st ex",df_ex1o['金额(元)'].sum())
print("outcome to 2st ex",df_ex2o['金额(元)'].sum())
print("outcome to dad and mom",df_momo['金额(元)'].sum() + df_dado['金额(元)'].sum())

income from 1st ex 1561.0
income from 2st ex 198.9
income from dad and mom 58742.66
outcome to 1st ex 3587.17
outcome to 2st ex 6395.0
outcome to dad and mom 0.0


## 1st Ex love me more than 2nd but I gave more money to 2nd ex. But you should know parents always love you mostly
    * People always love who love them not that much but ignore Who really love them
    * tips: stop spend money and time on who maybe will leave you. Spend more money and time with parents. They are getting older.